Add known issues around transit managed keys (#23080)
hashicorp/vault: website/content/partials/known-issues/transit-managed-keys-panics.mdx

### Transit Encryption with Cloud KMS managed keys causes a panic

#### Affected versions

- 1.13.1+ up to 1.13.7 inclusively

> - 1.14.0+ up to 1.14.3 inclusively.

Reviewer: ```suggestion
- 1.14.0+ up to 1.14.3 inclusively
```

Style correction: format all line items consistently

Author: Applied in https://github.com/hashicorp/vault/pull/23080/commits/73a206f089239be8eee21737a8db955c6bbe762b

#### Issue

> Vault will panic when a Transit encryption API call is backed by a Cloud KMS managed key (Azure, GCP, AWS).

Reviewer: ```suggestion
Vault panics when it receives a Transit encryption API call that is backed by a Cloud KMS managed key (Azure, GCP, AWS).
```

Style correction: write in active voice, make it clear who/what is taking the action

Author: Applied in https://github.com/hashicorp/vault/pull/23080/commits/73a206f089239be8eee21737a8db955c6bbe762b

> <Note>Encryption and decryption APIs are not affected with PKCS#11 managed keys and non-managed key, key types</Note>

Reviewer 2: ```suggestion
<Note>Encryption and decryption APIs are not affected with PKCS#11 managed keys and transit native keys</Note>
```

Author: Thanks you beat me to updating from cli.

Reviewer 2: Whoops :P

> <Note>Encryption and decryption APIs are not affected with PKCS#11 managed keys and non-managed key, key types</Note>

Reviewer: ```suggestion
<Note>

The issue does not affect encryption and decryption with the following key types:

- PKCS#11 managed keys
- non-managed keys

</Note>
```

Style correction: make important information easy to scan and parse

Author: Applied in https://github.com/hashicorp/vault/pull/23080/commits/73a206f089239be8eee21737a8db955c6bbe762b, just tweaked non-managed keys to Transit native keys
